Source/Core/DiscIO/DirectoryBlob.h

```diff
@@ -173,7 +176,8 @@ class DiscContentContainer
   template <typename T>
   void AddReference(u64 offset, const std::vector<T>& vector)
   {
-    return Add(offset, vector.size() * sizeof(T), reinterpret_cast<const u8*>(vector.data()));
+    return Add(offset, vector.size() * sizeof(T),
+               std::make_shared<std::vector<u8>>(vector.begin(), vector.end()));
```

Comment:
Change the type of `vector` to `std::vector<T>` (no `const` or `&`) and replace `std::vector<u8>>(vector.begin(), vector.end())` with `std::move(vector)`. This way you'll be able to skip a copy if someone is passing in an rvalue vector.

Reply:
Actually, on second thought, I'm not sure if this function is needed anymore. Could callers just call `Add` instead, or would that require them to manually call `std::make_shared`?

If you do keep this function, please remove `Reference` from the name. It's no longer adding a reference, it's adding a copy.